server/engine/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/cbl/CblParser.java

```diff
@@ -0,0 +1,384 @@
+        one("(");
+        String argument = next();
+        int i = checkIntegerLiteral(argument);
+        one(")");
+        directiveNodes.add(createDirectiveNode(start, pos));
+        removeSegments(start, pos);
+        if (isLineNumberWrong(i)) {
+          semanticError(argument, "LINECOUNT must be an integer between 1 and 255");
+        }
+      } else if (isNext("SPACE")) {
+        one("SPACE");
+        int start = pos - 1;
+        one("(");
+        String next = next();
+        int i = checkIntegerLiteral(next);
+        if (i != 1 && i != 2 && i != 3) {
+          semanticError(next, "SPACE must be 1, 2 or 3.");
+        }
+        one(")");
+        directiveNodes.add(createDirectiveNode(start, pos));
+        removeSegments(start, pos);
+      } else if (isNext("NATLANG")) {
+        one("NATLANG");
+        int start = pos - 1;
+        one("(");
+        or("CS", "EN", "KA");
+        one(")");
+        directiveNodes.add(createDirectiveNode(start, pos));
+        removeSegments(start, pos);
+      } else {
+        try {
+          or(SIMPLE_CICS_OPTIONS);
+          int start = pos - 1;
+          directiveNodes.add(createDirectiveNode(start, start + 1));
+          removeSegments(start, start + 1);
+        } catch (CblDiagnosticException e) {
+          hasNonCics = true;
+          if (force) {
+            throw e;
+          } else {
+            parseOption();
+          }
+        }
+      }
+      opt(",");
+    }
+    if (segments[pos].equals(")")) {
+      removePrevComma(optStart, pos);
+    }
+    if (inApostrophesOrQuotes) {
+      opt("'");
+      opt("\"");
+    }
+    one(")");
+    opt(",");
+    return hasNonCics;
+  }
+
+  private void removePrevComma(int from, int to) {
+    for (int i = to - 1; i >= from; i--) {
+      if (segments[i] == null || isBlank(segments[i])) {
+        continue;
+      }
+      if (",".equals(segments[i])) {
+        segments[i] = null;
+      }
+      return;
+    }
+  }
+
+  private void removeSegments(int start, int end) {
+    Arrays.fill(segments, start, end, null);
+  }
+
+  private CompilerDirectiveNode createDirectiveNode(int start, int end) {
+    StringBuilder text = new StringBuilder();
+    for (int i = start; i < end; i++) {
+      text.append(segments[i].trim());
+    }
+    Range range =
+        new Range(new Position(lineNumber, ranges[start]), new Position(lineNumber, ranges[end]));
+    Locality locality = Locality.builder().uri(uri).range(range).build();
+    return new CompilerDirectiveNode(locality, text.toString(), CICSDialect.DIALECT_NAME);
+  }
+
+  private void semanticError(String next, String message) {
+    diagnostics.add(
+        new CblDiagnosticException(makeCurrentLocation(), next, WARNING, message)
+            .toSyntaxError(lineNumber));
+  }
+
+  private boolean isLineNumberWrong(int nnn) {
+    // nnn must be an integer between 1 and 255.
+    if (nnn == 0) {
+      return true;
+    }
+    return nnn > 255;
```

Comment:
```suggestion
    return nnn <= 0 || nnn > 255;
```
`Integer.parseInt` will parse things like `-1`.